### PR TITLE
fix: KIP-368 use receiver goroutine to process all sasl v1 responses

### PR DESCRIPTION
**Pull request by k-wall**
Fixes defect (#2233) in kip-368 implementation where existing incoming wire traffic would get confused with re-authentication traffic leading to unexpected disconnects and OOM errors.

The approach taken uses the existing receiver routine to process all SASL v1+ responses for both authentication and re-authentication.   For SASL v0, the existing behaviour is maintained.

### CODE REVIEW

**Comment by ppatierno** [comment]
@dnwe do you think it is possible to review and get this fix merged soon? The issue makes the re-auth feature not usable.
Our next Strimzi canary release 0.3.0 was heavily based on this feature but now we are blocked waiting for the fix.
Any chance to have a Sarama patched release 1.33.1 out?

**Review by dnwe — APPROVED** [review]
LGTM

**Comment by dnwe** [comment]
@ppatierno no problem, I actually pushed it out as Version 1.34.0 because I wanted to include the KIP-345 (static membership) support to allow people to test that too

File: broker.go [hunk 1]
```diff
@@ -239,12 +244,27 @@ func (b *Broker) Open(conf *Config) error {
...
+		if conf.Net.SASL.Enable && !useSaslV0 {
+			b.connErr = b.authenticateViaSASLv1()
+			if b.connErr != nil {
+				close(b.responses)
+				err = b.conn.Close()
+				if err == nil {
+					DebugLogger.Printf("Closed connection to broker %s\n", b.addr)
+				} else {
+					Logger.Printf("Error while closing connection to broker %s: %s\n", b.addr, err)
+				}
```
Comment (pierDipi) [inline broker.go:257]: Hi @k-wall, I'm wondering if there is there a reason why `b.connErr` is not logged here too? https://github.com/IBM/sarama/issues/2994

**Pull request merged**